main.go

```diff
@@ -732,15 +723,17 @@ to get your bearings.
 		}
 		v := connect()
 		for _, path := range args {
-			if opt.Delete.Recurse {
+			_, key := vault.ParsePath(path)
+			//Ignore -R if path has a key because that makes no sense
+			if opt.Delete.Recurse && key == "" {
 				if !opt.Delete.Force && !recursively("delete", args...) {
 					return nil /* skip this command, process the next */
 				}
 				if err := v.DeleteTree(path); err != nil {
 					return err
 				}
 			} else {
-				if err := v.Delete(path); err != nil {
+				if err := v.Delete(path); err != nil && !vault.IsNotFound(err) {
```

Comment:
When I asked James, he decided it should work this way.
Also, it appears that the generic secret backend doesn't hand back a 404 or anything if you try to delete a secret that isn't present. That means checking for this requires an extra call to vault which... isn't detrimental and isn't a good reason. More of a design decision, I think. Maybe it should warn instead?